packages/service-library/src/servicelib/pools.py

```diff
@@ -0,0 +1,34 @@
+from concurrent.futures import ProcessPoolExecutor
+from contextlib import contextmanager
+
+# only gets created on use and is guaranteed to be the s
+# ame for the entire lifetime of the application
+__shared_process_pool_executor = None
+
+
+def get_shared_process_pool_executor(**kwargs) -> ProcessPoolExecutor:
+    global __shared_process_pool_executor  # pylint: disable=global-statement
+
+    if __shared_process_pool_executor is None:
+        # pylint: disable=consider-using-with
+        __shared_process_pool_executor = ProcessPoolExecutor(**kwargs)
+
+    return __shared_process_pool_executor
+
+
+@contextmanager
+def non_blocking_process_pool_executor(**kwargs) -> ProcessPoolExecutor:
+    """
+    Avoids default context manger behavior which calls
+    shutdown with wait=True an blocks.
+    """
+    executor = get_shared_process_pool_executor(**kwargs)
+    try:
+        yield executor
+    finally:
+        # due to an issue in cpython https://bugs.python.org/issue34073
+        # bypassing shutdown and using a shared pool
+        # remove call to get_shared_process_pool_executor and replace with
+        # a new instance when the issue is fixed
+        # executor.shutdown(wait=False)
+        pass
```

Comment:
this is worrisome ... having a resource that is not deallocated. This might bring other problems

I wonder, what is the advantage of this vs the default executor?

Reply:
When calling `loop.run_in_executor(None, coroutine())` a ThreadPool is used. In some cases a ProcessPool is required, usually when working with CPU bound tasks.

Reply:
Got it: So default executor is a threadpool 👍 

What about unclosed executor above?

Reply:
Since the lifecycle of the ProcessPoolExecutor object equals to the lifecycle of the application. I think it's fine to not close it, by basically bypassing the blocking issue.

The above solution was implemented because there is no proper way of closing the `ProcessPoolExecutor`. Not even by patching or hacking around what is causing the issue.

Reply:
OK. Please add a ``FIXME:`` note in the bug so it gets at least highlighted and we do not forget

Reply:
Done!